UnicodeDecodeError is raised while executing west build

**Describe the bug**
`UnicodeDecodeError `is raised while executing `west build`

>   File "zephyrproject\zephyr\scripts/west_commands\zcmake.py", line 217, in load
>     for line_no, line in enumerate(cache):
> UnicodeDecodeError: 'cp950' codec can't decode byte 0xe2 in position 1838: illegal multibyte sequence

It comes from` zephy/scripts/west_commands/zcmake.py`, line 217
I have to add `encoding="utf-8"` to line 216 to avoid this error

```
        with open(cache_file, 'r', encoding="utf-8") as cache:
            for line_no, line in enumerate(cache):
```

**To Reproduce**
Steps to reproduce the behavior:
1. cd zephyr\samples\bluetooth\peripheral_hr
2. west build -b nrf52_pca10040

**Screenshots or console output**

```
Traceback (most recent call last):
  File "c:\python37\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "c:\python37\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Python37\Scripts\west.exe\__main__.py", line 9, in <module>
  File "c:\python37\lib\site-packages\west\_bootstrap\main.py", line 499, in main
    wrap(wrap_argv)
  File "c:\python37\lib\site-packages\west\_bootstrap\main.py", line 485, in wrap
    west.main.main(argv)
  File "zephyrproject\.west\west\src\west\main.py", line 580, in main
    args.handler(args, unknown)
  File "zephyrproject\.west\west\src\west\main.py", line 332, in ext_command_handler
    command.run(*west_parser.parse_known_args(argv))
  File "zephyrproject\.west\west\src\west\commands\command.py", line 90, in run
    self.do_run(args, unknown)
  File "zephyrproject\zephyr\scripts\west_commands\build.py", line 135, in do_run
    self._setup_build_dir()
  File "zephyrproject\zephyr\scripts\west_commands\build.py", line 231, in _setup_build_dir
    source_dir=source_dir, app=app)
  File "zephyrproject\zephyr\scripts/west_commands\build_helpers.py", line 95, in find_build_dir
    if default and is_zephyr_build(default):
  File "zephyrproject\zephyr\scripts/west_commands\build_helpers.py", line 115, in is_zephyr_build
    cache = zcmake.CMakeCache.from_build_dir(path)
  File "zephyrproject\zephyr\scripts/west_commands\zcmake.py", line 208, in from_build_dir
    return CMakeCache(os.path.join(build_dir, DEFAULT_CACHE))
  File "zephyrproject\zephyr\scripts/west_commands\zcmake.py", line 212, in __init__
    self.load(cache_file)
  File "zephyrproject\zephyr\scripts/west_commands\zcmake.py", line 217, in load
    for line_no, line in enumerate(cache):
UnicodeDecodeError: 'cp950' codec can't decode byte 0xe2 in position 1838: illegal multibyte sequence
```

**Environment (please complete the following information):**
 - OS: Windows 10 Traditional Chinese
 - Toolchain: gnuarmemb
 - Commit SHA: 765c063
 - Terminal: cmd.exe

Funnily I get a different issue:

```
C:\Users\Carles\src\tmp\我能吞下玻璃而不伤身体\zephyr>west build -b reel_boards samples\hello_world
-- west build: build configuration:
       source directory: C:\Users\Carles\src\tmp\我能吞下玻璃而不伤身体\zephyr\samples\hello_world
       build directory: C:\Users\Carles\src\tmp\我能吞下玻璃而不伤身体\zephyr\build (created)
       BOARD: reel_boards (origin: command line)
-- west build: generating a build system
CMake Error at CMakeLists.txt:5 (include):
  include could not find load file:

    C:\Users\Carles\src\tmp\???????????\zephyr/cmake/app/boilerplate.cmake


-- The C compiler identification is GNU 7.1.0
-- The CXX compiler identification is GNU 7.1.0
-- Check for working C compiler: C:/Strawberry/c/bin/gcc.exe
-- Check for working C compiler: C:/Strawberry/c/bin/gcc.exe -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - failed
-- Check for working CXX compiler: C:/Strawberry/c/bin/c++.exe
-- Check for working CXX compiler: C:/Strawberry/c/bin/c++.exe -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - failed
CMake Error at CMakeLists.txt:8 (target_sources):
  Cannot specify sources for target "app" which is not built by this project.


-- Configuring incomplete, errors occurred!
See also "C:/Users/Carles/src/tmp/我能吞下玻璃而不伤身体/zephyr/build/CMakeFiles/CMakeOutput.log".
See also "C:/Users/Carles/src/tmp/我能吞下玻璃而不伤身体/zephyr/build/CMakeFiles/CMakeError.log".
ERROR: command exited with status 1: 'C:\Program Files\CMake\bin\cmake.EXE' '-BC:\Users\Carles\src\tmp\我能吞下玻璃而不伤身体\zephyr\build' '-SC:\Users\Carles\src\tmp\我能吞下玻璃而不伤身体\zephyr\samples\hello_world' -GNinja -DBOARD=reel_boards
run as "west -v build -b reel_boards 'samples\hello_world'" for a stack trace
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -459,7 +459,8 @@ add_subdirectory(drivers)
 
 # Include zephyr modules generated CMake file.
 if(EXISTS ${CMAKE_BINARY_DIR}/zephyr_modules.txt)
-  file(STRINGS ${CMAKE_BINARY_DIR}/zephyr_modules.txt ZEPHYR_MODULES_TXT)
+	file(STRINGS ${CMAKE_BINARY_DIR}/zephyr_modules.txt ZEPHYR_MODULES_TXT
+		ENCODING UTF-8)
 
   foreach(module ${ZEPHYR_MODULES_TXT})
     # Match "<name>":"<path>" for each line of file, each corresponding to
@@ -517,7 +518,7 @@ execute_process(
   --trigger          ${syscalls_subdirs_trigger} # Trigger file that is used for json generation
   ${syscalls_links}                              # If defined, create symlinks for dependencies
 )
-file(STRINGS ${syscalls_subdirs_txt} PARSE_SYSCALLS_PATHS_DEPENDS)
+file(STRINGS ${syscalls_subdirs_txt} PARSE_SYSCALLS_PATHS_DEPENDS ENCODING UTF-8)
 
 if(${CMAKE_HOST_SYSTEM_NAME} STREQUAL Windows)
   # On windows only adding/removing files or folders will be reflected in depends.

--- a/scripts/subfolder_list.py
+++ b/scripts/subfolder_list.py
@@ -63,14 +63,14 @@ def main():
     existing = ''
 
     if os.path.exists(args.out_file):
-        with open(args.out_file, 'r') as fp:
+        with open(args.out_file, 'r', encoding="utf-8") as fp:
             existing = fp.read()
 
         if new != existing:
-            with open(args.out_file, 'w') as fp:
+            with open(args.out_file, 'w', encoding="utf-8") as fp:
                 fp.write(new)
     else:
-        with open(args.out_file, 'w') as fp:
+        with open(args.out_file, 'w', encoding="utf-8") as fp:
             fp.write(new)
 
     # Always touch trigger file to ensure json files are updated

--- a/scripts/west_commands/zcmake.py
+++ b/scripts/west_commands/zcmake.py
@@ -213,7 +213,7 @@ class CMakeCache:
 
     def load(self, cache_file):
         entries = []
-        with open(cache_file, 'r') as cache:
+        with open(cache_file, 'r', encoding="utf-8") as cache:
             for line_no, line in enumerate(cache):
                 entry = CMakeCacheEntry.from_line(line, line_no)
                 if entry:

--- a/scripts/zephyr_module.py
+++ b/scripts/zephyr_module.py
@@ -147,10 +147,10 @@ def main():
         projects += args.extra_modules
 
     if args.kconfig_out:
-        kconfig_out_file = open(args.kconfig_out, 'w')
+        kconfig_out_file = open(args.kconfig_out, 'w', encoding="utf-8")
 
     if args.cmake_out:
-        cmake_out_file = open(args.cmake_out, 'w')
+        cmake_out_file = open(args.cmake_out, 'w', encoding="utf-8")
 
     try:
         for project in projects:

